File: app/src/main/java/mrkm4ntr/twitterclient/activities/DetailActivity.java
package mrkm4ntr.twitterclient.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import mrkm4ntr.twitterclient.R;
import mrkm4ntr.twitterclient.views.DetailFragment;

public class DetailActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        if (savedInstanceState == null) {
            Bundle arguments = new Bundle();
            arguments.putParcelable(DetailFragment.DETAIL_URI, getIntent().getData());
            DetailFragment fragment = new DetailFragment();
            fragment.setArguments(arguments);

            getSupportFragmentManager().beginTransaction()
                    .add(R.id.status_detail_container, fragment)
                    .commit();
        }
    }

}

File: app/src/main/kotlin/mrkm4ntr/twitterclient/activities/DetailActivity.kt
package mrkm4ntr.twitterclient.activities

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.Menu

import mrkm4ntr.twitterclient.R
import mrkm4ntr.twitterclient.views.DetailFragment

class DetailActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)
        val toolbar = findViewById(R.id.toolbar) as Toolbar
        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        if (savedInstanceState == null) {
            val arguments = Bundle()
            arguments.putParcelable(DetailFragment.DETAIL_URI, intent.data)
            val fragment = DetailFragment()
            fragment.arguments = arguments

            supportFragmentManager.beginTransaction().add(R.id.status_detail_container, fragment).commit()
        }
    }

}
